modules/Installer/AWS.Tools.Installer.psm1

```diff
@@ -501,40 +466,60 @@ function Install-AWSToolsModule {
         }
 
         if ($Name -eq 'AWS.Tools.Installer') {
             throw "AWS.Tools.Installer cannot be used to install AWS.Tools.Installer. Use Update-Module instead."
         }
-        
-        if ($RequiredVersion) {
-            $versionToInstall = $RequiredVersion
+
+        [PSObject[]]$availableModulesToInstall = Find-AWSToolsModule -Name $Name -Proxy $Proxy -ProxyCredential $ProxyCredential
+
+        [System.Version]$availableVersion = [System.Version[]]$availableModulesToInstall.Version | Measure-Object -Minimum | Select-Object -Expand Minimum
+
+        if ($MinimumVersion -and $MinimumVersion -gt $availableVersion) {
+            throw "The maximum version available is $availableVersion."
         }
-        else {
-            $versionToInstall = Get-AvailableModuleVersion -Name $Name -MaximumVersion $MaximumVersion -Proxy $Proxy -ProxyCredential $ProxyCredential
-            Write-Verbose "[$($MyInvocation.MyCommand)] Installing AWS Tools version $versionToInstall"
+        if ($RequiredVersion -and $RequiredVersion -gt $availableVersion) {
+            throw "The maximum version available is $availableVersion."
+        }
+        if ($MinimumVersion -and $RequiredVersion -and $MinimumVersion -gt $RequiredVersion) {
+            throw "Parameter MinimumVersion is greater than RequiredVersion."
+        }
+        if ($MaximumVersion -and $RequiredVersion -and $MaximumVersion -lt $RequiredVersion) {
+            throw "Parameter MaximumVersion is less than RequiredVersion."
+        }
+        if ($MaximumVersion -and -not $RequiredVersion -and $MaximumVersion -lt $availableVersion) {
+            $RequiredVersion = Find-Module -Name 'AWS.Tools.Common' -MaximumVersion $MaximumVersion | Select-Object -Expand Version
+        }
+        if (-not $RequiredVersion) {
+            $RequiredVersion = $availableVersion
         }
 
+        Write-Verbose "[$($MyInvocation.MyCommand)] Installing AWS Tools version $RequiredVersion"
+
+        [string[]]$modulesToInstall = $availableModulesToInstall.Name
+
         if (-not $SkipUpdate) {
             Write-Verbose "[$($MyInvocation.MyCommand)] Searching installed modules"
-            $awsToolsModules = Get-AWSToolsModule -SkipIfInvalidSignature
-            if ($awsToolsModules) {
-                Write-Verbose "[$($MyInvocation.MyCommand)] Merging existing modules into the list of modules to install"
-                $Name = ($Name + ($awsToolsModules | Select-Object -Expand Name)) | Sort-Object -Unique
+            [System.Management.Automation.PSModuleInfo[]]$installedAwsToolsModules = Get-AWSToolsModule -SkipIfInvalidSignature
+            if ($installedAwsToolsModules) {
+                $modulesToInstall = ($modulesToInstall + ($installedAwsToolsModules | Select-Object -Expand Name)) | Sort-Object -Unique
+                Write-Verbose "[$($MyInvocation.MyCommand)] Merging existing modules into the list of modules to install: ($modulesToInstall)"
```

Comment:
I am taking the approach of using `string.Join` for normally visible output because it is sightly more pleasant to the reader to have comma-separated lists. I am instead simply printing the list variable for verbose output in order to keep the code from being overly complicated.